common/djangoapps/course_modes/views.py

```diff
@@ -135,6 +136,7 @@ def get(self, request, course_id, error=None):
 
         return render_to_response("course_modes/choose.html", context)
 
+    @method_decorator(transaction.non_atomic_requests)
     @method_decorator(login_required)
```

Comment:
Right - I got application order and execution order mixed up. Order seems right.